src/plans/D3MAaveDaiPlan.sol

```diff
@@ -0,0 +1,202 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Copyright (C) 2021-2022 Dai Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity 0.6.12;
+
+import "./ID3MPlan.sol";
+
+interface TokenLike {
+    function totalSupply() external view returns (uint256);
+    function balanceOf(address) external view returns (uint256);
+}
+
+interface LendingPoolLike {
+    function getReserveData(address asset) external view returns (
+        uint256,    // Configuration
+        uint128,    // the liquidity index. Expressed in ray
+        uint128,    // variable borrow index. Expressed in ray
+        uint128,    // the current supply rate. Expressed in ray
+        uint128,    // the current variable borrow rate. Expressed in ray
+        uint128,    // the current stable borrow rate. Expressed in ray
+        uint40,     // last updated timestamp
+        address,    // address of the adai interest bearing token
+        address,    // address of the stable debt token
+        address,    // address of the variable debt token
+        address,    // address of the interest rate strategy
+        uint8       // the id of the reserve
+    );
+}
+
+interface InterestRateStrategyLike {
+    function OPTIMAL_UTILIZATION_RATE() external view returns (uint256);
+    function EXCESS_UTILIZATION_RATE() external view returns (uint256);
+    function variableRateSlope1() external view returns (uint256);
+    function variableRateSlope2() external view returns (uint256);
+    function baseVariableBorrowRate() external view returns (uint256);
+    function getMaxVariableBorrowRate() external view returns (uint256);
+}
+
+contract D3MAaveDaiPlan is ID3MPlan {
+
+    LendingPoolLike          public immutable pool;
+    InterestRateStrategyLike public immutable interestStrategy;
+    TokenLike                public immutable stableDebt;
+    TokenLike                public immutable variableDebt;
+    address                  public immutable adai;
+    address                  public immutable dai;
```

Comment:
If we use TokenLike for the other immutables we should use it for `dai` as well,

Reply:
done. I did not change adai to `TokenLike` since it is never used as a contract.  let me know if you feel strongly that consistency is important for that one